test/gtest/conv_immediate.cpp

```diff
@@ -0,0 +1,532 @@
+                                            desc.GetConvStrides(),
+                                            desc.GetConvDilations(),
+                                            desc.GetGroupCount());
+
+            EXPECT_TRUE(miopenConvolutionBackwardWeightsImmediate(
+                            &handle,
+                            &out.descriptor,
+                            out_dev.get(),
+                            &in.descriptor,
+                            in_dev.get(),
+                            convDesc,
+                            &wei.descriptor,
+                            wei_dev.get(),
+                            nullptr,
+                            0,
+                            miopen::solver::Id("ConvDirectNaiveConvWrw").Value()) ==
+                        miopenStatusSuccess);
+
+            wei.read_gpu_data(handle, wei_dev);
+
+            // we expect exact match, since use integer
+            valid_result = verify_tensor(wei.check, wei.host);
+        }
+        break;
+        }
+
+        EXPECT_TRUE(valid_result);
+
+        miopenDestroyConvolutionDescriptor(convDesc);
+    };
+};
+
+template <class Types>
+class ReferenceConv2d : public ReferenceConvBase<Types>
+{
+protected:
+    void run() { this->iterate_conv_2d(this->run_conv_nd); }
+};
+
+// clang-format off
+#define MakeTypeDefs(layout)                                                                \
+    TypeDefs<miopen::conv::Direction::Forward,          float,      float,      layout>,    \
+    TypeDefs<miopen::conv::Direction::Forward,          float16,    float16,    layout>,    \
+    TypeDefs<miopen::conv::Direction::Forward,          bfloat16,   bfloat16,   layout>,    \
+    TypeDefs<miopen::conv::Direction::Forward,          int8_t,     int32_t,    layout>,    \
+    TypeDefs<miopen::conv::Direction::Forward,          int8_t,     float,      layout>,    \
+                                                                                            \
+    TypeDefs<miopen::conv::Direction::BackwardData,     float,      float,      layout>,    \
+    TypeDefs<miopen::conv::Direction::BackwardData,     float16,    float16,    layout>,    \
+    TypeDefs<miopen::conv::Direction::BackwardData,     bfloat16,   bfloat16,   layout>,    \
+                                                                                            \
+    TypeDefs<miopen::conv::Direction::BackwardWeights,  float,      float,      layout>,    \
+    TypeDefs<miopen::conv::Direction::BackwardWeights,  float16,    float16,    layout>,    \
+    TypeDefs<miopen::conv::Direction::BackwardWeights,  bfloat16,   bfloat16,   layout>
+
+// clang-format on
+
+using Implementations2d =
+    ::testing::Types<MakeTypeDefs(miopenTensorNCHW), MakeTypeDefs(miopenTensorNHWC)>;
+
+TYPED_TEST_CASE(ReferenceConv2d, Implementations2d);
+
+TYPED_TEST(ReferenceConv2d, Test) { this->run(); }
+
+template <class Types>
+struct ReferenceConv3d : ReferenceConvBase<Types>
```

Comment:
Do these tests cover the same functionality as [gpu_reference_kernel.cpp](https://github.com/ROCm/MIOpen/blob/da4ba7f838c8cd97e5a59e12c873f060e78d627c/test/gpu_reference_kernel.cpp)?

If yes, then should we remove the gpu_reference_kernel tests with this change?

Reply:
> Do these tests cover the same functionality as [gpu_reference_kernel.cpp](https://github.com/ROCm/MIOpen/blob/da4ba7f838c8cd97e5a59e12c873f060e78d627c/test/gpu_reference_kernel.cpp)?
> 
> If yes, then should we remove the gpu_reference_kernel tests with this change?

Yes, this is a direct replacement/port to gtest. I've just kept the original for reference temporarily.